### PR TITLE
Backdate Alabama parameter values to 2021

### LINKED ISSUE #3333: Backdate Alabama parameter values to 2021
Used for taxsim validation

### PR DESCRIPTION
**Pull request by XZ-mabel**
Fixes #3333

### Summary
📝📈🗺️

This change updates the changelog entry and the Alabama parameter values for the policy engine. It is a minor version bump that reflects the latest data sources for the state-level policies.

> _`changelog` updated_
> _minor bump, new parameters_
> _springtime for Alabama_

### Walkthrough
*  Bump the minor version and update the changelog entry ([link](https://github.com/PolicyEngine/policyengine-us/pull/3367/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R4))

### CODE REVIEW

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: need a git pull upstream master

File: policyengine_us/variables/gov/states/al/tax/income/deductions/al_federal_income_tax_deduction.py [hunk 1]
```diff
@@ -10,5 +10,7 @@ class al_federal_income_tax_deduction(Variable):
     reference = "https://www.revenue.alabama.gov/ultraviewer/viewer/basic_viewer/index.html?form=2023/01/22f40abk.pdf#page=20"
     defined_for = StateCode.AL
 
-    adds = "gov.states.al.tax.income.deductions.federal_tax.countable_sources"
-    subtracts = "gov.states.al.tax.income.deductions.federal_tax.credits"
+    adds = [
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/al/tax/income/deductions/al_federal_income_tax_deduction.py:13]: we do not need the squared brackets here, only for variable inputs, same for the subtracts

File: policyengine_us/parameters/gov/states/al/tax/income/deductions/standard/phase_out/threshold.yaml [hunk 2]
```diff
@@ -8,16 +8,24 @@ metadata:
   reference:
   - title: The Code of Alabama 1975 Section 40-18-15(b)(4)
     href: https://alisondb.legislature.state.al.us/alison/CodeOfAlabama/1975/Coatoc.htm
+  - title: The Code of Alabama 1975 Section 40-18-15(b)(3)
+    href: https://alisondb.legislature.state.al.us/alison/CodeOfAlabama/1975/Coatoc.htm
   - title: 2022 Alabama Department of Revenue -Form 40 Booklet Page9
     href: https://www.revenue.alabama.gov/ultraviewer/viewer/basic_viewer/index.html?form=2023/01/22f40bk.pdf
-
+  - title: 2021 Alabama Department of Revenue -Form 40 Booklet Page9
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/al/tax/income/deductions/standard/phase_out/threshold.yaml:15]:
```suggestion
  - title: 2021 Alabama Department of Revenue -Form 40 Booklet Page 9
```

File: policyengine_us/parameters/gov/states/al/tax/income/deductions/standard/phase_out/threshold.yaml [hunk 3]
```diff
@@ -8,16 +8,24 @@ metadata:
...
+    href: https://www.revenue.alabama.gov/wp-content/uploads/2022/06/21f40abk.pdf
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/al/tax/income/deductions/standard/phase_out/threshold.yaml:16]:
```suggestion
    href: https://www.revenue.alabama.gov/wp-content/uploads/2022/06/21f40abk.pdf#page=9
```

File: policyengine_us/parameters/gov/states/al/tax/income/deductions/standard/phase_out/rate.yaml [hunk 4]
```diff
@@ -8,16 +8,20 @@ metadata:
   reference:
   - title: The Code of Alabama 1975 Section 40-18-15(b)(4)
     href: https://alisondb.legislature.state.al.us/alison/CodeOfAlabama/1975/Coatoc.htm
+  - title: The Code of Alabama 1975 Section 40-18-15(b)(3)
+    href: https://alisondb.legislature.state.al.us/alison/CodeOfAlabama/1975/Coatoc.htm
   - title: 2022 Alabama Department of Revenue -Form 40 Booklet Page9
     href: https://www.revenue.alabama.gov/ultraviewer/viewer/basic_viewer/index.html?form=2023/01/22f40bk.pdf
+  - title: 2021 Alabama Department of Revenue -Form 40 Booklet Page9
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/al/tax/income/deductions/standard/phase_out/rate.yaml:15]:
```suggestion
  - title: 2021 Alabama Department of Revenue -Form 40 Booklet Page 9
```

File: policyengine_us/parameters/gov/states/al/tax/income/deductions/standard/phase_out/rate.yaml [hunk 5]
```diff
@@ -8,16 +8,20 @@ metadata:
...
+    href: https://www.revenue.alabama.gov/wp-content/uploads/2022/06/21f40abk.pdf
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/al/tax/income/deductions/standard/phase_out/rate.yaml:16]:
```suggestion
    href: https://www.revenue.alabama.gov/wp-content/uploads/2022/06/21f40abk.pdf#page=9
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**